File: formRestoreDatabase.vb
Public Class formRestoreDatabase
    Private mSettings As New IniFile

    Private Class SettingsClass
        Public Property ID As String
        Public Property Name As String
    End Class


    Private Sub Me_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        groupboxSQLServerInfo.Visible = False
        groupboxFoldersAndFiles.Visible = False
        groupboxActions.Visible = False

        comboboxSettings.ValueMember = "ID"
        comboboxSettings.DisplayMember = "Name"

        ListSettings()
    End Sub

    Private Sub ListSettings()
        Dim INIPathAndFileName As String = My.Application.Info.DirectoryPath & "\" & AppConstants.SETTINGS_FILENAME

        comboboxSettings.Items.Clear()

        If My.Computer.FileSystem.FileExists(INIPathAndFileName) Then
            If mSettings.Read(INIPathAndFileName, False) Then
                For Each SectionCurrent As CSSQLDBBackupTools.Section In mSettings.GetSections
                    ' Add new Settings object to combobox
                    Dim SettingsEntry As New SettingsClass

                    SettingsEntry.ID = SectionCurrent.Name  ' Section ID

                    ' Find the Section Name
                    Dim NameKey As CSSQLDBBackupTools.Key
                    NameKey = mSettings.GetKey("Name", SectionCurrent)
                    If Not NameKey Is Nothing AndAlso NameKey.Value.Trim <> "" Then
                        SettingsEntry.Name = NameKey.Value.Trim
                    Else
                        comboboxSettings.Items.Add(SectionCurrent.Name)
                    End If
                    comboboxSettings.Items.Add(SettingsEntry)
                Next
                comboboxSettings.Enabled = True
            Else
                comboboxSettings.Items.Add("Settings file could not be read.")
                comboboxSettings.SelectedIndex = 0
                comboboxSettings.Enabled = False

                groupboxSQLServerInfo.Visible = False
                groupboxFoldersAndFiles.Visible = False
                groupboxActions.Visible = False
            End If
        Else
            comboboxSettings.Items.Add("Settings file not found on app folder.")
            comboboxSettings.SelectedIndex = 0
            comboboxSettings.Enabled = False

            groupboxSQLServerInfo.Visible = False
            groupboxFoldersAndFiles.Visible = False
            groupboxActions.Visible = False
        End If
    End Sub

    Private Sub Settings_SelectedIndexChanged(sender As Object, e As EventArgs) Handles comboboxSettings.SelectedIndexChanged
        Dim SettingsCurrent As SettingsClass
        Dim KeyCurrent As CSSQLDBBackupTools.Key

        ' Muestro todos los datos de la Sección
        groupboxSQLServerInfo.Visible = True
        groupboxFoldersAndFiles.Visible = True
        groupboxActions.Visible = True

        SettingsCurrent = CType(comboboxSettings.SelectedItem, SettingsClass)

        ' SQL Server Instance Name
        KeyCurrent = mSettings.GetKey("SQLServerInstanceName", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            textboxSQLServerInstanceName.Text = KeyCurrent.Value.Trim
        Else
            textboxSQLServerInstanceName.Text = ""
        End If

        ' SQL Server Login
        KeyCurrent = mSettings.GetKey("SQLServerLogin", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            textboxSQLServerLogin.Text = KeyCurrent.Value.Trim
        Else
            textboxSQLServerLogin.Text = ""
        End If

        ' SQL Server Password
        KeyCurrent = mSettings.GetKey("SQLServerPassword", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            Dim PasswordDecrypter As New CS_Encrypt_TripleDES(CardonerSistemas.Constants.PublicEncryptionPassword)
            Dim DecryptedPassword As String = ""
            If Not PasswordDecrypter.Decrypt(KeyCurrent.Value.Trim, DecryptedPassword) Then
                textboxSQLServerPassword.Text = ""
            Else
                textboxSQLServerPassword.Text = DecryptedPassword
            End If
            PasswordDecrypter = Nothing
        Else
            textboxSQLServerPassword.Text = ""
        End If

        ' SQL Server Database
        KeyCurrent = mSettings.GetKey("SQLServerDatabase", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            textboxSQLServerDatabase.Text = KeyCurrent.Value.Trim
        Else
            textboxSQLServerDatabase.Text = ""
        End If

        ' Backup Folder
        KeyCurrent = mSettings.GetKey("BackupFolder", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            textboxBackupFolder.Text = KeyCurrent.Value.Trim
        Else
            textboxBackupFolder.Text = ""
        End If

        ' Backup FileName
        KeyCurrent = mSettings.GetKey("BackupFileName", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            textboxBackupFileName.Text = KeyCurrent.Value.Trim
        Else
            textboxBackupFileName.Text = ""
        End If

        ' Compressed FileName
        KeyCurrent = mSettings.GetKey("CompressedFileName", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            textboxCompressedFileName.Text = KeyCurrent.Value.Trim
        Else
            textboxCompressedFileName.Text = ""
        End If

        ' Restore Folder
        KeyCurrent = mSettings.GetKey("RestoreFolder", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            textboxRestoreFolder.Text = KeyCurrent.Value.Trim
        Else
            textboxRestoreFolder.Text = ""
        End If

        ' Restore Data FileName
        KeyCurrent = mSettings.GetKey("RestoreDataFileName", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            textboxRestoreDataFileName.Text = KeyCurrent.Value.Trim
        Else
            textboxRestoreDataFileName.Text = ""
        End If

        ' Restore Log FileName
        KeyCurrent = mSettings.GetKey("RestoreLogFileName", SettingsCurrent.ID)
        If Not KeyCurrent Is Nothing AndAlso KeyCurrent.Value.Trim <> "" Then
            textboxRestoreLogFileName.Text = KeyCurrent.Value.Trim
        Else
            textboxRestoreLogFileName.Text = ""
        End If

        ' Clear the most recent file information
        textboxMostRecentBackupFileName.Text = ""
        textboxMostRecentBackupFileModifiedDateTime.Text = ""
    End Sub

    Private Sub BackupFolder_Click(sender As Object, e As EventArgs) Handles buttonBackupFolder.Click
        If textboxBackupFolder.Text.Trim <> "" Then
            FolderBrowserDialog1.SelectedPath = textboxBackupFolder.Text.Trim
        Else
            FolderBrowserDialog1.RootFolder = Environment.SpecialFolder.MyDocuments
        End If

        Dim result As DialogResult = FolderBrowserDialog1.ShowDialog(Me)
        If result = Windows.Forms.DialogResult.OK Then
            textboxBackupFolder.Text = FolderBrowserDialog1.SelectedPath
        End If
    End Sub

    Private Sub RestoreFolder_Click(sender As Object, e As EventArgs) Handles buttonRestoreFolder.Click
        If textboxRestoreFolder.Text.Trim <> "" Then
            FolderBrowserDialog1.SelectedPath = textboxRestoreFolder.Text.Trim
        Else
            FolderBrowserDialog1.RootFolder = Environment.SpecialFolder.MyDocuments
        End If

        Dim result As DialogResult = FolderBrowserDialog1.ShowDialog(Me)
        If result = Windows.Forms.DialogResult.OK Then
            textboxRestoreFolder.Text = FolderBrowserDialog1.SelectedPath
        End If
    End Sub

    Private Sub FindMostRecentBackupFile_Click(sender As Object, e As EventArgs) Handles buttonFindMostRecentBackupFile.Click
        If textboxBackupFolder.Text.Trim = "" Then
            MsgBox("Backup folder is not specified.", MsgBoxStyle.Information, My.Application.Info.Title)
            textboxBackupFolder.Focus()
            Exit Sub
        End If
        If Not My.Computer.FileSystem.DirectoryExists(textboxBackupFolder.Text.Trim) Then
            MsgBox("Backup folder specified doesn't exists.", MsgBoxStyle.Exclamation, My.Application.Info.Title)
            textboxBackupFolder.Focus()
            Exit Sub
        End If
        If textboxBackupFileName.Text.Trim = "" Then
            MsgBox("Backup filename is not specified.", MsgBoxStyle.Information, My.Application.Info.Title)
            textboxBackupFileName.Focus()
            Exit Sub
        End If
        If textboxCompressedFileName.Text.Trim = "" Then
            MsgBox("Compressed filename is not specified.", MsgBoxStyle.Information, My.Application.Info.Title)
            textboxCompressedFileName.Focus()
            Exit Sub
        End If

        ' Construct the Search Pattern
        Dim SearchPattern As String
        Dim IndexOfFirstSpecialChar As Integer
        Dim IndexOfExtensionDot As Integer

        IndexOfFirstSpecialChar = textboxCompressedFileName.Text.Trim.IndexOf("{")
        If IndexOfFirstSpecialChar + 1 = textboxCompressedFileName.Text.Trim.Length Then
            ' Compressed File Name doesn't contains special char
            SearchPattern = textboxCompressedFileName.Text.Trim
        Else
            SearchPattern = textboxCompressedFileName.Text.Trim.Substring(0, IndexOfFirstSpecialChar)
            IndexOfExtensionDot = textboxCompressedFileName.Text.Trim.LastIndexOf(".")
            If IndexOfExtensionDot > 0 Then
                SearchPattern &= "*" & textboxCompressedFileName.Text.Trim.Substring(IndexOfExtensionDot)
            End If
        End If

        Dim di As New IO.DirectoryInfo(textboxBackupFolder.Text.Trim)
        Dim fi As IO.FileInfo

        fi = di.GetFiles(SearchPattern, IO.SearchOption.TopDirectoryOnly).OrderByDescending(Function(f) f.LastWriteTime).FirstOrDefault

        If fi Is Nothing Then
            textboxMostRecentBackupFileName.Text = ""
            textboxMostRecentBackupFileModifiedDateTime.Text = ""
        Else
            textboxMostRecentBackupFileName.Text = fi.Name
            textboxMostRecentBackupFileModifiedDateTime.Text = fi.LastWriteTime.ToShortDateString & " " & fi.LastWriteTime.ToShortTimeString
        End If

    End Sub

    Private Sub RestoreDatabase_Click(sender As Object, e As EventArgs) Handles buttonRestoreDatabase.Click
        If textboxMostRecentBackupFileName.Text = "" Then
            MsgBox("You must find the most recent Backup file before restore the database.", MsgBoxStyle.Information, My.Application.Info.Title)
            buttonFindMostRecentBackupFile.Focus()
            Exit Sub
        End If

        Dim BackupFolderAndCompressedFileName As String
        Dim RestoreFolderAndExtractedBackupFileName As String
        Dim IndexOfExtensionDot As Integer

        ' Get folders and files names
        BackupFolderAndCompressedFileName = textboxBackupFolder.Text.Trim & CStr(IIf(textboxBackupFolder.Text.Trim.Last = "\", "", "\")) & textboxMostRecentBackupFileName.Text
        RestoreFolderAndExtractedBackupFileName = textboxRestoreFolder.Text.Trim & CStr(IIf(textboxRestoreFolder.Text.Trim.Last = "\", "", "\")) & textboxMostRecentBackupFileName.Text
        IndexOfExtensionDot = textboxMostRecentBackupFileName.Text.Trim.LastIndexOf(".")
        If IndexOfExtensionDot > 0 Then
            RestoreFolderAndExtractedBackupFileName = RestoreFolderAndExtractedBackupFileName.Substring(0, IndexOfExtensionDot) & RestoreFolderAndExtractedBackupFileName.Substring(IndexOfExtensionDot).Replace(".7z", ".bak")
        End If

        Me.Cursor = Cursors.WaitCursor
        Application.DoEvents()

        ' Decompress the backup file
        Dim psiDecompress As New ProcessStartInfo
        psiDecompress.FileName = COMPRESS_APPLICATION_7ZIP_FILENAME
        psiDecompress.Arguments = String.Format(COMPRESS_APPLICATION_7ZIP_DECOMPRESS_ARGUMENTS, BackupFolderAndCompressedFileName, textboxRestoreFolder.Text)
        psiDecompress.UseShellExecute = True
        psiDecompress.WindowStyle = ProcessWindowStyle.Hidden

        Dim processDecompress As Process = Process.Start(psiDecompress)
        Do While Not processDecompress.HasExited
            Application.DoEvents()
        Loop

        If processDecompress.ExitCode = 0 Then
            Try
                ' Connect to SQL Server and execute the restore statement
                Dim conn As New SqlClient.SqlConnection
                conn.ConnectionString = String.Format(SQLSERVER_CONNECTIONSTRING, textboxSQLServerInstanceName.Text.Trim, "master", My.Settings.SQL_ConnectionTimeout, textboxSQLServerLogin.Text.Trim, textboxSQLServerPassword.Text.Trim)
                conn.Open()
                Dim comm As SqlClient.SqlCommand = conn.CreateCommand
                comm.CommandTimeout = My.Settings.SQL_CommandTimeout
                comm.CommandText = String.Format(SQLSERVER_RESTORE_STATEMENT, textboxSQLServerDatabase.Text.Trim, RestoreFolderAndExtractedBackupFileName, textboxRestoreFolder.Text.Trim & CStr(IIf(textboxRestoreFolder.Text.Trim.Last = "\", "", "\")), textboxRestoreDataFileName.Text.Trim, textboxRestoreLogFileName.Text.Trim)
                comm.CommandType = CommandType.Text
                comm.ExecuteNonQuery()

                ' Delete the decompressed backup file
                My.Computer.FileSystem.DeleteFile(RestoreFolderAndExtractedBackupFileName)
                Me.Cursor = Cursors.Default
                Application.DoEvents()
                MsgBox("Database restored succesfully.", MsgBoxStyle.Information, My.Application.Info.Title)
                Exit Try

            Catch ex As Exception
                Me.Cursor = Cursors.Default
                Application.DoEvents()
                MsgBox("Error restoring database." & vbCrLf & vbCrLf & "Error: " & ex.Message, MsgBoxStyle.Critical, My.Application.Info.Title)
            End Try
        Else
            Me.Cursor = Cursors.Default
            Application.DoEvents()
        End If
    End Sub
End Class

File: FormRestoreDatabase.cs
using CardonerSistemas.Framework.Base;
using System.Data;
using System.Data.SqlClient;
using System.Diagnostics;

namespace CSSqlDbBackupTools
{
    public partial class FormRestoreDatabase : Form
    {

        #region Declarations

#pragma warning disable S1144 // Unused private types or members should be removed
#pragma warning disable IDE0051 // Remove unused private members

        private const string CompressApplicationFilename = "7Z.exe";
        private const string CompressApplicationCompressArguments = "a -t7z \"{0}\" \"{1}\" -mx=9";
        private const string CompressApplicationDecompressArguments = "e -aoa \"{0}\" -o\"{1}\"";
        private const string CompressedFileExtension = "7z";

        private const string BackupFileExtension = "bak";

        private const string SqlServerConnectionString = "Data Source={0};Database={1};Connection Timeout={2};User ID={3};Password={4};";
        private const string SqlServerBackupStatement = "BACKUP DATABASE [{0}] TO DISK = {1} WITH COPY_ONLY, NOFORMAT, INIT, NAME = N'{0} - Full Copy-only Backup', SKIP, NOREWIND, NOUNLOAD, STATS = 10";
        private const string SqlServerRestoreStatement = "RESTORE DATABASE [{0}] FILE = N'{0}' FROM DISK = N'{1}' WITH  FILE = 1, MOVE N'{0}' TO N'{2}{3}', MOVE N'{0}_log' TO N'{2}{4}', NOUNLOAD, REPLACE, STATS = 10";
        private const string SqlServerMasterDatabaseName = "master";

#pragma warning restore IDE0051 // Remove unused private members
#pragma warning restore S1144 // Unused private types or members should be removed

        #endregion

        public FormRestoreDatabase()
        {
            InitializeComponent();
        }

        private void This_Load(object sender, EventArgs e)
        {
            GroupBoxSqlServerInfo.Visible = false;
            GroupBoxFoldersAndFiles.Visible = false;
            GroupBoxActions.Visible = false;

            ComboBoxSettings.DisplayMember = "Name";

            ListSettings();
        }

        private void ListSettings()
        {

            ComboBoxSettings.Items.Clear();

            foreach (Database database in Program.Settings.Databases)
            {
                if (string.IsNullOrWhiteSpace(database.Name))
                {
                    database.Name = "« Empty name »";
                }
                ComboBoxSettings.Items.Add(database);
            }
            ComboBoxSettings.Enabled = true;
        }

        private void ComboBoxSettings_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (ComboBoxSettings.SelectedItem is string)
            {
                return;
            }

            GroupBoxSqlServerInfo.Visible = true;
            GroupBoxFoldersAndFiles.Visible = true;
            GroupBoxActions.Visible = true;

            Database database = (Database)ComboBoxSettings.SelectedItem;
            if (database != null)
            {
                TextBoxSqlServerInstanceName.Text = database.SqlServerInstanceName;
                TextBoxSqlServerLogin.Text = database.SqlServerLogin;
                if (CardonerSistemas.Framework.Cryptography.StringCipher.Decrypt(database.SqlServerPassword, Program.PublicEncryptionPassword, out string decryptedPassword))
                {
                    TextBoxSqlServerPassword.Text = decryptedPassword;
                }
                else
                {
                    TextBoxSqlServerPassword.Text = string.Empty;
                }
                TextBoxSqlServerDatabase.Text = database.SqlServerDatabase;
                TextBoxBackupFolder.Text = database.BackupFolder;
                TextBoxBackupFileName.Text = database.BackupFileName;
                TextBoxCompressedFileName.Text = database.CompressedFileName;
                TextBoxRestoreFolder.Text = database.RestoreFolder;
                TextBoxRestoreDataFileName.Text = database.RestoreDataFileName;
                TextBoxRestoreLogFileName.Text = database.RestoreLogFileName;
            }
        }

        private void ButtonBackupFolder_Click(object sender, EventArgs e)
        {
            // TODO:
        }

        private void ButtonRestoreFolder_Click(object sender, EventArgs e)
        {
            // TODO:
        }

        private void ButtonFindMostRecentBackupFile_Click(object sender, EventArgs e)
        {
            if (TextBoxBackupFolder.Text.Trim() == string.Empty)
            {
                MessageBox.Show("Backup folder not specified.", CardonerSistemas.Framework.Base.Application.Info.Title, MessageBoxButtons.OK, MessageBoxIcon.Information);
                return;
            }
            if (!Path.Exists(TextBoxBackupFolder.Text.Trim()))
            {
                MessageBox.Show("Backup folder doesn't exist.", CardonerSistemas.Framework.Base.Application.Info.Title, MessageBoxButtons.OK, MessageBoxIcon.Information);
                return;
            }
            if (TextBoxBackupFileName.Text.Trim() == string.Empty)
            {
                MessageBox.Show("Backup file name not specified.", CardonerSistemas.Framework.Base.Application.Info.Title, MessageBoxButtons.OK, MessageBoxIcon.Information);
                return;
            }
            if (TextBoxCompressedFileName.Text.Trim() == string.Empty)
            {
                MessageBox.Show("Compressed file name not specified.", CardonerSistemas.Framework.Base.Application.Info.Title, MessageBoxButtons.OK, MessageBoxIcon.Information);
                return;
            }

            string searchPattern;

            int indexOfFirstSpecialChar = TextBoxCompressedFileName.Text.Trim().IndexOf('{');
            if (indexOfFirstSpecialChar == -1)
            {
                searchPattern = TextBoxCompressedFileName.Text.Trim();
            }
            else
            {
                searchPattern = TextBoxCompressedFileName.Text.Trim()[..indexOfFirstSpecialChar];
                int indexOfExtensionDot = TextBoxCompressedFileName.Text.Trim().LastIndexOf('.');
                if (indexOfExtensionDot > 0)
                {
                    searchPattern += string.Concat("*", TextBoxCompressedFileName.Text.Trim().AsSpan(indexOfExtensionDot));
                }
            }

            FileInfo fileInfo = new DirectoryInfo(TextBoxBackupFolder.Text.Trim()).GetFiles(searchPattern, SearchOption.TopDirectoryOnly).OrderByDescending(fileInfo => fileInfo.LastWriteTime).FirstOrDefault();
            if (fileInfo == null)
            {
                TextBoxMostRecentBackupFileName.Text = string.Empty;
                TextBoxMostRecentBackupFileModifiedDateTime.Text = string.Empty;
            }
            else
            {
                TextBoxMostRecentBackupFileName.Text = fileInfo.Name;
                TextBoxMostRecentBackupFileModifiedDateTime.Text = $"{fileInfo.LastWriteTime.ToShortDateString()} {fileInfo.LastWriteTime.ToShortTimeString()}";
            }
        }

        private void ButtonRestoreDatabase_Click(object sender, EventArgs e)
        {
            if (TextBoxMostRecentBackupFileName.Text == string.Empty)
            {
                MessageBox.Show("You must find the most recente backup file before restore the database.", CardonerSistemas.Framework.Base.Application.Info.Title, MessageBoxButtons.OK, MessageBoxIcon.Information);
                return;
            }

            string backupFolderAndCompressedFileName = Path.Combine(TextBoxBackupFolder.Text, TextBoxMostRecentBackupFileName.Text);
            string restoreFolderAndExtractedBackupFileName = Path.Combine(TextBoxRestoreFolder.Text, Path.GetFileNameWithoutExtension(TextBoxMostRecentBackupFileName.Text) + '.' + BackupFileExtension);

            Application.DoEvents();
            this.Cursor = Cursors.WaitCursor;

            // Decompress the backup file
            try
            {
                System.Diagnostics.Process processDecompress = System.Diagnostics.Process.Start(new ProcessStartInfo()
                {
                    FileName = CompressApplicationFilename,
                    Arguments = string.Format(CompressApplicationDecompressArguments, backupFolderAndCompressedFileName, TextBoxRestoreFolder.Text),
                    UseShellExecute = true,
                    WindowStyle = ProcessWindowStyle.Hidden
                });
                processDecompress.WaitForExit(Program.Settings.ProcessTimeoutMilliseconds);

                if (processDecompress.ExitCode != 0)
                {
                    this.Cursor = Cursors.Default;
                    return;
                }
            }
            catch (Exception ex)
            {
                Error.ProcessException(ex, "Error decompressing the compressed backup file.");
                return;
            }

            // Restore the backup file to SQL Server
            try
            {
                // Connect to SQL Server and execute the restore statement
                SqlConnection sqlConnection = new(string.Format(SqlServerConnectionString, TextBoxSqlServerInstanceName.Text.Trim(), SqlServerMasterDatabaseName, Program.Settings.SqlConnectionTimeout, TextBoxSqlServerLogin.Text.Trim(), TextBoxSqlServerPassword.Text.Trim()));
                sqlConnection.Open();
                SqlCommand sqlCommand = sqlConnection.CreateCommand();
                sqlCommand.CommandTimeout = Program.Settings.SqlConnectionTimeout;
                sqlCommand.CommandText = string.Format(SqlServerRestoreStatement, TextBoxSqlServerDatabase.Text.Trim(), restoreFolderAndExtractedBackupFileName, FileSystem.PathAddDirectorySeparator(TextBoxRestoreFolder.Text), TextBoxRestoreDataFileName.Text.Trim(), TextBoxRestoreLogFileName.Text.Trim());
                sqlCommand.CommandType = CommandType.Text;
                sqlCommand.ExecuteNonQuery();
            }
            catch (Exception ex)
            {
                this.Cursor = Cursors.Default;
                Error.ProcessException(ex, "Error restoring the database backup.");
                return;
            }

            // Delete the decompressed backup file
            try
            {
                File.Delete(restoreFolderAndExtractedBackupFileName);
            }
            catch (Exception ex)
            {
                this.Cursor = Cursors.Default;
                Error.ProcessException(ex, "Error deleting backup file.");
                return;
            }

            this.Cursor = Cursors.Default;
            MessageBox.Show("Database has been restored succesfully.", CardonerSistemas.Framework.Base.Application.Info.Title, MessageBoxButtons.OK, MessageBoxIcon.Information);
        }

    }
}
